### PR TITLE
Markerclustering really large dataset

### PR DESCRIPTION
**Pull request by timvanoostrom**
In this PR I've done the following:

- Added a more representational dataset
- Added a story for filtering clustered datasets 
- Added some improvements to the MarkerClusterGroup component

### CODE REVIEW

File: stories/src/examples/MarkerFilters.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        isActive: false,
+      })
+    }
+    return datasets
+  }, [markers])
+
+  const markersFiltered = useMemo(() => {
+    let i = 0
+    // eslint-disable-next-line no-shadow
+    let markersFiltered: any[] = []
```
Comment (chrisvanmook) [inline stories/src/examples/MarkerFilters.tsx:48]: We could use `LatLngTuple` from leaflet as a type here

File: stories/src/examples/MarkerFilters.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import React, { ReactNode, useMemo, useState } from 'react'
+import { Label, Checkbox, themeColor } from '@datapunt/asc-ui'
+import styled from 'styled-components'
+
+interface MarkerFiltersProps {
+  markers: Array<[number, number]>
```
Comment (chrisvanmook) [inline stories/src/examples/MarkerFilters.tsx:6]: This should be of type LatLngTuple[]

File: stories/src/examples/MarkerFilters.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    }
+    return datasets
+  }, [markers])
+
+  const markersFiltered = useMemo(() => {
+    let i = 0
+    // eslint-disable-next-line no-shadow
+    let markersFiltered: any[] = []
+    const setLen = markers.length / setCount
+    for (i; i < activeDatasets.length; i += 1) {
```
Comment (chrisvanmook) [inline stories/src/examples/MarkerFilters.tsx:50]: Also why not define `i` directly here?

File: stories/src/examples/MarkerFilters.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    <Panel>
+      Total: {markersFiltered.length}
+      <UnstyledOl>
+        {markerSets.map((markerSet) => (
+          <li key={markerSet.title}>
+            <Label htmlFor={markerSet.title} label={markerSet.title}>
+              <Checkbox
+                id={markerSet.title}
+                checked={activeDatasets.includes(markerSet.id)}
+                onChange={() => {
```
Comment (chrisvanmook) [inline stories/src/examples/MarkerFilters.tsx:69]: Could you move this to a defined method, and use the `useCallback` hook?

**Review by chrisvanmook — CHANGES_REQUESTED** [review]

File: stories/src/examples/MarkerFilters.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+const UnstyledOl = styled.ol`
+  margin: 0;
+  padding: 0;
+  list-style-type: none;
+`
+
+const Panel = styled.div`
+  background-color: ${themeColor('tint', 'level2')};
+  padding: 10px;
+  box-shadow: 0 0 0 5px rgba(0, 0, 0, 0.1);
```
[the remark this answers is not in the document]
Comment (timvanoostrom) [reply stories/src/examples/MarkerFilters.tsx:19]: Ah, yeah. This is just an ad-hoc style. The story is more about filtering and the underlying functionality. I can use 2px here, no problem!

File: stories/src/examples/MarkerFilters.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  markers,
+  clusterComponent,
+}: MarkerFiltersProps) {
+  const [activeDatasets, setActiveDataset] = useState([0, 1, 2])
+
+  const markerSets = useMemo(() => {
+    const datasets = []
+    let i = 0
+
+    for (i; i < setCount; i += 1) {
```
[the remark this answers is not in the document]
Comment (timvanoostrom) [reply stories/src/examples/MarkerFilters.tsx:34]: Old habit.. I'll change it.

File: stories/src/examples/MarkerFilters.tsx (hunk 1 above)
Comment (timvanoostrom) [reply stories/src/examples/MarkerFilters.tsx:48]: Will do.

**Review by chrisvanmook — APPROVED** [review]

**Pull request merged**